summarization-api-playground/src/main.ts

```diff
@@ -64,26 +71,37 @@ const initializeApplication = async () => {
   }
 
   let timeout: number | undefined = undefined;
-  inputTextArea.addEventListener('input', () => {
-    characterCountSpan.textContent = inputTextArea.value.length.toFixed();
-    if (inputTextArea.value.length > MAX_MODEL_CHARS) {
-      characterCountSpan.classList.add('tokens-exceeded');
-      characterCountExceededSpan.classList.remove('hidden');
-    } else {
-      characterCountSpan.classList.remove('tokens-exceeded');
-      characterCountExceededSpan.classList.add('hidden');
-    }
-
+  function scheduleSummarization() {
     // Debounces the call to the summarization API. This will run the summarization once the user
     // hasn't typed anything for at least 1 second.
     clearTimeout(timeout);
     timeout = setTimeout(async () => {
       output.textContent = 'Generating summary...';
-      let session = await createSummarizationSession();
+      let session = await createSummarizationSession(
+        summaryTypeSelect.value as AISummarizerType,
+        summaryFormatSelect.value as AISummarizerFormat,
+        summaryLengthSelect.value as AISummarizerLength,
+      );
       let summary = await session.summarize(inputTextArea.value);
       session.destroy();
       output.textContent = summary;
     }, 1000);
+  }
+
+  summaryTypeSelect.addEventListener('change', scheduleSummarization);
+  summaryFormatSelect.addEventListener('change', scheduleSummarization);
+  summaryLengthSelect.addEventListener('change', scheduleSummarization);
+
+  inputTextArea.addEventListener('input', () => {
+    characterCountSpan.textContent = inputTextArea.value.length.toFixed();
```

Comment:
```suggestion
    characterCountSpan.textContent = inputTextArea.value.length;
```

Maybe use [`NumberFormat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat) here, but it's probably overkill.

Reply:
Typescript doesn't like removing `toFixed()` because `textContent` is `string` but `inputTextArea.value` is `number`, so a conversion is needed. This is the error: `Type 'number' is not assignable to type 'string'.`